### PR TITLE
feat(components/modals): add `SkyConfirmTestingController` to mock confirm dialogs in tests

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2114/storybook/)

Component Storybooks:

- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/2114/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2114/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/2114/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/2114/storybooks/avatar-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2114/storybooks/data-manager-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/2114/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/2114/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/2114/storybooks/forms-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/2114/storybooks/layout-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2114/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/2114/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2114/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2114/storybooks/phone-field-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/2114/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2114/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2114/storybooks/text-editor-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2114/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/2114/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2114/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2114/playground/)

File: apps/code-examples/src/app/code-examples/modals/confirm/demo-controller.spec.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    fixture.componentInstance.launchConfirm();
+    fixture.detectChanges();
+
+    confirmController.expectOpen({
+      message: 'Are you sure?',
+    });
+
+    confirmController.ok();
+    confirmController.expectNone();
```
Comment (Blackbaud-SteveBrush) [inline apps/code-examples/src/app/code-examples/modals/confirm/demo-controller.spec.ts:64]: ☝️ Here are the key pieces.

File: libs/components/modals/testing/src/confirm/confirm-testing.controller.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  public abstract expectNone(): void;
+  /**
+   * Throws if the open confirm dialog does not match the provided configuration.
+   * @param config
+   */
+  public abstract expectOpen(config: SkyConfirmConfig): void;
+  /**
+   * Closes the confirm dialog with the provided action.
+   */
+  public abstract close(args: SkyConfirmCloseEventArgs): void;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/modals/testing/src/confirm/confirm-testing.controller.ts:25]: We have the `ok` and `cancel` methods to handle the supported actions; the `close` method is used for custom button actions.

File: libs/components/modals/testing/src/confirm/confirm-testing.service.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  return;
+}
+
+/**
+ * @internal
+ */
+export class SkyConfirmTestingService
+  implements SkyConfirmServiceInterface, SkyConfirmTestingController
```
Comment (Blackbaud-PaulCrowder) [inline libs/components/modals/testing/src/confirm/confirm-testing.service.ts]: You're implementing a class like it's an interface, but classes should be extended.

**Review by Blackbaud-PaulCrowder — APPROVED** [review]

**Pull request merged**